src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/mobile/get_interior_vehicle_data_consent_request.cc

```diff
@@ -118,8 +139,31 @@ void GetInteriorVehicleDataConsentRequest::on_event(
     return;
   }
 
-  const auto& hmi_response = event.smart_object();
+  auto temp_response = event.smart_object();
+
+  LOG4CXX_DEBUG(logger_,
+                "Adding back filtered out module ids for response to mobile");
+
+  smart_objects::SmartObject all_consents;
+  GetLocationConsents(all_consents);
+
+  const auto response_consents =
+      temp_response[app_mngr::strings::msg_params][message_params::kAllowed]
+          .asArray();
+  uint32_t response_consents_counter = 0;
+
+  for (auto& consent : *(all_consents.asArray())) {
+    // Only modify consent for moduleIds allowed by location constraints
+    if (consent.asBool()) {
+      consent = (*response_consents)[response_consents_counter];
```

Comment:
This line segfaults core when the hmi responds with an error case. Allowed param does not exist in this case and dereferencing the null ptr crashes core.

Reply:
Fixed in https://github.com/smartdevicelink/sdl_core/pull/3039/commits/cba159f9fe3fed8c13808c242d05133706d395d0